grammar.js

```diff
@@ -261,31 +258,24 @@ module.exports = grammar({
       optional('export'),
       'type',
       optional('rec'),
-      $._type_declaration
+      sep1(
+        seq(repeat($._newline), repeat($.decorator), 'and'),
+        $.type_binding
+      )
     ),
 
-    _type_declaration: $ => seq(
-      choice($.type_identifier, $.type_identifier_path),
+    type_binding: $ => seq(
+      field('name', choice($.type_identifier, $.type_identifier_path)),
       optional($.type_parameters),
       optional(seq(
-        optional(seq('=', $._type)),
+        optional(seq('=', $._non_function_inline_type)),
```

Comment:
This change is to ensure that the match after the type equation has a field `body`.